src/runtime/manifest-parser.peg

```diff
@@ -327,16 +327,27 @@ ParticleClaimDerivesFrom
   }
 
 ParticleCheckStatement
-  = 'check' whiteSpace handle:lowerIdent whiteSpace expression:ParticleCheckExpressionBody eolWhiteSpace
+  = 'check' whiteSpace target:ParticleCheckTarget whiteSpace expression:ParticleCheckExpressionBody eolWhiteSpace
   {
     return {
       kind: 'particle-trust-check',
       location: location(),
-      handle,
+      target,
       expression,
     } as AstNode.ParticleCheckStatement;
   }
 
+ParticleCheckTarget
+  = name:lowerIdent isSlot:(whiteSpace 'data')?
+  {
+    return {
+      kind: 'particle-check-target',
+      location: location(),
+      targetType: isSlot ? 'slot' : 'handle',
```

Comment:
Is this going to turn into a type check later? Do slot and handle have different rules for checking?

Reply:
Not in the near future. And no, the checking rules for slots and handles are both very similar. They both essentially turn into the same types of nodes in the data flow graph.